libs/payments/customer/src/lib/invoice.factories.ts

```diff
@@ -20,6 +20,15 @@ export const InvoicePreviewFactory = (
     faker.string.alphanumeric({ length: 8 }).toLocaleUpperCase() +
     '-' +
     faker.string.numeric({ length: 4, allowLeadingZeros: true }),
-  ...override,
   nextInvoiceDate: faker.date.future().getDate(),
+  ...override,
+});
+
+export const InvoicePreviewForUpgradeFactory = (
+  override?: Partial<InvoicePreviewForUpgrade>
+): InvoicePreviewForUpgrade => ({
+  ...InvoicePreviewFactory(override),
+  oneTimeCharge: faker.number.int({ max: 1000 }),
+  oneTimeChargeSubtotal: faker.number.int({ max: 1000 }),
```

Comment:
If either of these returns a value of 0, it may cause unexpected behavior in tests. Can you add in a `min` value?